File: project_euler/057/A.py
def iter(frac):
    a = [frac[1], frac[0]+frac[1]]
    return [a[0]+a[1], a[1]]


FRAC = [1, 1]
NUM = 0
for _ in range(1000):
    FRAC = iter(FRAC)
    if len(str(FRAC[0])) > len(str(FRAC[1])):
        NUM += 1

print NUM

File: project_euler/057/A.js
"use strict";

var gcd = function (a, b) {
  if (b === 0) {
    return a;
  }
  return gcd(b, a % b);
};

var simplify = function (frac) {
  var fGcd = gcd(frac[0], frac[1]);
  return [frac[0]/fGcd, frac[1]/fGcd];
};

var iter = function (frac) {
  var n = frac[0];
  var d = frac[1];
  var a = [d, n+d];
  return [a[0]+a[1], a[1]];
};

var frac = [1, 1];
var i;
var num = 0;
for (i = 0; i < 1000; i += 1) {
  frac = iter(frac);
  if (frac[0].toString().length > frac[1].toString().length) {
    num += 1;
  }
  console.log(i +': '+ frac+' : '+num);
}
console.log(num);
